compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala

```diff
@@ -261,11 +261,11 @@ class ClassfileParser(
           addConstructorTypeParams(denot)
         }
 
-        denot.info = cook.apply(pool.getType(in.nextChar))
+        denot.info = pool.getType(in.nextChar)
         if (isEnum) denot.info = ConstantType(Constant(sym))
         if (isConstructor) normalizeConstructorParams()
         setPrivateWithin(denot, jflags)
-        denot.info = translateTempPoly(parseAttributes(sym, denot.info))
+        denot.info = translateTempPoly(cook.apply(parseAttributes(sym, denot.info)))
```

Comment:
With the change, the two applications of `cook.apply` are in 1-to-1 correspondence with the two calls to `parseAttributes`. Add the `cook.apply` to the result type of `parseAttributes` instead? 

Otherwise LGTM

Reply:
Makes sense, PR updated.